rust/lance-index/src/scalar/btree.rs

```diff
@@ -831,6 +829,44 @@ impl ScalarIndex for BTreeIndex {
             store,
         )?))
     }
+
+    async fn remap(
+        &self,
+        mapping: &HashMap<u64, Option<u64>>,
+        dest_store: &dyn IndexStore,
+    ) -> Result<()> {
+        // Remap and write the pages
+        let mut sub_index_file = dest_store
+            .new_index_file(BTREE_PAGES_NAME, self.sub_index.schema().clone())
+            .await?;
+
+        let sub_index_reader = self.store.open_index_file(BTREE_PAGES_NAME).await?;
+
+        for page_number in self.page_lookup.all_page_ids() {
+            let old_serialized = sub_index_reader.read_record_batch(page_number).await?;
+            let remapped = self
+                .sub_index
+                .remap_subindex(old_serialized, mapping)
+                .await?;
+            sub_index_file.write_record_batch(remapped).await?;
+        }
+
+        sub_index_file.finish().await?;
+
+        // Copy the lookup file as-is
+        let lookup_reader = self.store.open_index_file(BTREE_LOOKUP_NAME).await?;
+        let lookup_as_batch = lookup_reader.read_record_batch(0).await?;
+
+        let mut new_lookup_writer = dest_store
+            .new_index_file(BTREE_LOOKUP_NAME, lookup_as_batch.schema())
+            .await?;
+        new_lookup_writer
+            .write_record_batch(lookup_as_batch)
+            .await?;
+        new_lookup_writer.finish().await?;
```

Comment:
Does the file actually need to be rewritten, or can we copy the underlying file?

Reply:
We just need to copy the file.  Mainly I was trying to avoid making the `IndexStore` abstraction too complicated.  Right now it is just "open file for reading batches" and "open file for writing batches" but I can add a "copy file to another store" as well since it will probably be a common enough operation.

Reply:
I've added a `copy_index_file` function which will just copy the underlying file